modules/git/repo_compare.go

```diff
@@ -263,3 +268,40 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatch will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatch(prID int64) (commitSHA string) {
+	// Migrated repositories download patches to "pulls" location
+	patchFile := fmt.Sprintf("pulls/%d.patch", prID)
+	loadPatch, err := ioutil.ReadFile(filepath.Join(repo.Path, patchFile))
+	if err != nil {
+		logger.Error("Could not read file %v, %v", patchFile, err)
+		return ""
+	}
+	// Parse the Patch stats, sometimes Migration returns a 404 for the patch file
+	commitSHAGroups := patchCommits.FindStringSubmatch(string(loadPatch))
```

Comment:
This operation maybe inefficient.

Reply:
I changed the function to return an error more cleanly, if you agree ?